src/inet/UDPEndPointImplSockets.cpp

```diff
@@ -274,6 +274,9 @@ CHIP_ERROR UDPEndPointImplSockets::ListenImpl()
 
 CHIP_ERROR UDPEndPointImplSockets::SendMsgImpl(const IPPacketInfo * aPktInfo, System::PacketBufferHandle && msg)
 {
+    // Ensure packet buffer is not null
+    VerifyOrReturnError(!msg.IsNull(), CHIP_ERROR_NO_MEMORY);
```

Comment:
Should this check also be done in src/inet/UDPEndPoint.cpp ?

Here, null message means 'invalid argument' not 'no memory'. Allocating it in UDPEndpoint or any other caller would be no memory.

Reply:
Looking at call stacks, it looks like SendMsg also just passes a buffer around.

I believe 'NO_MEMORY' should be returned as a result of an allocation failure. Here it seems just the packet is null in an input argument that should never be null. that is `CHIP_ERROR_INVALID_ARGUMENT`

Reply:
Sure I can agree with that argument to change the return status from being NO_MEMORY to being invalid argument.

Before I change the status code, do you agree with the conversation below with @bzbarsky-apple  about moving this check from the SendMsgImpl to being inside the UDPEndPoint::SendMsg function?

Reply:
@andy31415  I've changed the return value from CHIP_ERROR_NO_MEMORY --> CHIP_ERROR_INVALID_ARGUMENT.